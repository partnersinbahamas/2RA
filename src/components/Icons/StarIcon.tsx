import React from 'react';
import PropTypes from 'prop-types';
import { BLACK } from '../Colors/colors';
import Svg from './Svg';

type TProps = {
  backgroundColor?: string;
  borderColor?: string;
  className?: string;
  color?: string;
  isHalfFilled?: boolean;
  style?: { [cssAttribute: string]: string };
  withBox?: boolean;
};

const StarIcon: React.FC<TProps> = ({
  backgroundColor,
  borderColor,
  className,
  color,
  isHalfFilled,
  style,
  withBox,
}) => {
  const gradient = isHalfFilled && (
    <defs>
      <linearGradient id="gradient">
        <stop offset="50%" stopColor={color} />
        <stop offset="50%" stopColor={backgroundColor} />
      </linearGradient>
    </defs>
  );
  return (
    <Svg
      xmlns="http://www.w3.org/2000/svg"
      viewBox={withBox ? '10.5 713 32 32' : '14.5 717 24 24'}
      style={style}
      className={className}
    >
      {gradient}
      <g>
        <polygon
          style={{
            fill: gradient ? 'url(#gradient)' : color,
            stroke: borderColor,
          }}
          points="31.5,730.5 38.5,725.5 29.5,725.5 26.5,716.5 23.5,725.5 14.5,725.5 21.5,730.5 18.5,739.5 
			26.5,733.9004 34.5,739.5"
        />
      </g>
    </Svg>
  );
};

StarIcon.propTypes = {
  backgroundColor: PropTypes.string,
  borderColor: PropTypes.string,
  className: PropTypes.string,
  color: PropTypes.string,
  isHalfFilled: PropTypes.bool,
  style: PropTypes.any,
  withBox: PropTypes.bool,
};

StarIcon.defaultProps = {
  borderColor: 'transparent',
  className: '',
  color: BLACK,
  style: {},
  withBox: false,
};

export default StarIcon;
