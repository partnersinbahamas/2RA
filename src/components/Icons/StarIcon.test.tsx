import React from 'react';
import { render } from '@testing-library/react';
import StarIcon from './StarIcon';

describe('StarIcon', () => {
  it('renders', () => {
    const { container } = render(<StarIcon />);
    expect(container).toMatchSnapshot();
  });

  it('renders half-filled', () => {
    const { container } = render(
      <StarIcon
        backgroundColor="white"
        color="red"
        borderColor="black"
        isHalfFilled
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
