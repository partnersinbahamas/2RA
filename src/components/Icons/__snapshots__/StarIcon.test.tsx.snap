// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarIcon renders 1`] = `
<div>
  <svg
    class="sc-beySPh jroDkm"
    viewBox="14.5 717 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g>
      <polygon
        points="31.5,730.5 38.5,725.5 29.5,725.5 26.5,716.5 23.5,725.5 14.5,725.5 21.5,730.5 18.5,739.5  26.5,733.9004 34.5,739.5"
        style="fill: #000000; stroke: transparent;"
      />
    </g>
  </svg>
</div>
`;

exports[`StarIcon renders half-filled 1`] = `
<div>
  <svg
    class="sc-beySPh jroDkm"
    viewBox="14.5 717 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <defs>
      <lineargradient
        id="gradient"
      >
        <stop
          offset="50%"
          stop-color="red"
        />
        <stop
          offset="50%"
          stop-color="white"
        />
      </lineargradient>
    </defs>
    <g>
      <polygon
        points="31.5,730.5 38.5,725.5 29.5,725.5 26.5,716.5 23.5,725.5 14.5,725.5 21.5,730.5 18.5,739.5  26.5,733.9004 34.5,739.5"
        style="fill: url(#gradient); stroke: black;"
      />
    </g>
  </svg>
</div>
`;
